lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -93,6 +93,27 @@ OpFoldResult ConstantOp::fold(ArrayRef<Attribute> operands) {
 // Binary Operators
 //===----------------------------------------------------------------------===//
 
+OpFoldResult AddPrimOp::fold(ArrayRef<Attribute> operands) {
+  /// Any folding here requires a bitwidth extension.
+
+  /// If both operands are constant, and < 64 bits, then perform constant
+  /// folding. Cannot use constFoldBinaryOp, since the width of the constant is
+  /// different from operands.
+  if (operands[0] && operands[1] &&
+      lhs().getType().cast<FIRRTLType>().getBitWidthOrSentinel() < 64) {
+    auto op1 = operands[0].dyn_cast<IntegerAttr>();
+    auto op2 = operands[1].dyn_cast<IntegerAttr>();
```

Comment:
These two things can be `cast<>`'s.  We know that AddPrimOp takes firrtl integer operands, and this is the only thing that can come in here.